code/_onclick/hud/radial.dm

```diff
@@ -261,7 +261,11 @@ GLOBAL_LIST_EMPTY(radial_menus)
 	if(!uniqueid)
 		uniqueid = "defmenu_[user.UID()]_[anchor.UID()]"
 
-	if(GLOB.radial_menus[uniqueid])
+	if(GLOB.radial_menus[uniqueid]) // Calls the close button an already existing radial menu.
+		var/datum/radial_menu/existing_menu = GLOB.radial_menus[uniqueid]
+		existing_menu.finished = TRUE
+		qdel(existing_menu)
+		GLOB.radial_menus -= uniqueid
```

Comment:
What does this do to the first call of show_radial_menu that initialized this? It will probably try to qdel a null, won't it? I think you should just set `existing_menu.finished = TRUE` to finished and stop there and let the code down on lines 285 handle deletion of the radial

Reply:
I'll move the check for an existing menu down and refactor it to meet that standard, a moment.